llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1844,23 +1854,30 @@ class VPWidenInductionRecipe : public VPHeaderPHIRecipe {
 class VPWidenIntOrFpInductionRecipe : public VPWidenInductionRecipe {
   TruncInst *Trunc;
 
+  bool isUnrolled() const { return getNumOperands() == 6; }
+
 public:
   VPWidenIntOrFpInductionRecipe(PHINode *IV, VPValue *Start, VPValue *Step,
-                                VPValue *VF, const InductionDescriptor &IndDesc,
-                                DebugLoc DL)
+                                VPValue *VF, VPValue *StepVector,
```

Comment:
From what I can tell we only ever initialise this with a poison value, at least from looking at all the call sites in this patch. If it's always going to be poison then maybe we can just remove the extra argument and just always call `addOperand(PoisonValue::get(...));`?